src/rng.cpp

```diff
@@ -1,19 +1,30 @@
-#include "output.h"
 #include "rng.h"
-#include <stdlib.h>
+#include <random>
 
-long rng(long val1, long val2)
+namespace {
+std::mt19937 random_generator {std::random_device{}()};
+} //namespace
+
+void seed_random_number_generator(char const* seed)
+{
+    random_generator.seed(
+        static_cast<unsigned>(djb2_hash(reinterpret_cast<unsigned char const*>(seed))));
+}
+
+long rng(long const val1, long const val2)
 {
-    long minVal = (val1 < val2) ? val1 : val2;
-    long maxVal = (val1 < val2) ? val2 : val1;
-    return minVal + long((maxVal - minVal + 1) * double(rand() / double(RAND_MAX + 1.0)));
+    return std::uniform_int_distribution<long> {
```

Comment:
I'd rather not second-guess what the implementations are doing, it means if they change for the better we won't pick the changes up.